modules/image-loading-optimization/module.php

```diff
@@ -0,0 +1,326 @@
+<?php
+namespace Elementor\Modules\ImageLoadingOptimization;
+
+use Elementor\Core\Base\Module as BaseModule;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+class Module extends BaseModule {
+	/**
+	 * @var int Minimum square-pixels threshold.
+	 */
+	private $min_priority_img_pixels = 50000;
+
+	/**
+	 * @var int The number of content media elements to not lazy-load.
+	 */
+	private $omit_threshold = 3;
+
+	/**
+	 * @var array Keep a track of images for which loading optimization strategy were computed.
+	 */
+	private static $image_visited = [];
+
+	/**
+	 * Get Module name.
+	 */
+	public function get_name() {
+		return 'image-loading-optimization';
+	}
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		// Stop wp core logic.
+		add_action( 'init', [ $this, 'stop_core_fetchpriority_high_logic' ] );
+		add_filter( 'wp_lazy_loading_enabled', '__return_false' );
+
+		// Run optimization logic on header.
+		add_action( 'get_header', [ $this, 'set_buffer' ] );
+		// Ensure buffer is flushed (if any) before the content logic.
+		add_filter( 'the_content', [ $this, 'flush_header_buffer' ], 0 );
+
+		// Run optimization logic on content.
+		add_filter( 'wp_content_img_tag', [ $this, 'loading_optimization_image' ] );
+
+		// Run optimization logic on footer. Flushing of footer buffer will be handled by PHP script end default logic.
+		add_action( 'get_footer', [ $this, 'set_buffer' ] );
+	}
+
+	/**
+	 * Stop WordPress core fetchpriority logic by setting the wp_high_priority_element_flag flag to false.
+	 */
+	public function stop_core_fetchpriority_high_logic() {
+		// wp_high_priority_element_flag was only introduced in 6.3.0
+		if ( function_exists( 'wp_high_priority_element_flag' ) ) {
+			wp_high_priority_element_flag( false );
+		}
+	}
+
+	/**
+	 * Set buffer to handle header and footer content.
+	 */
+	public function set_buffer() {
+		ob_start( [ $this, 'handle_buffer_content' ] );
+	}
+
+	/**
+	 * This function ensure that buffer if any is flushed before the content is called.
+	 * This function behaves more like an action than a filter.
+	 *
+	 * @param string $content the content.
+	 * @return string We simply return the content from parameter.
+	 */
+	public function flush_header_buffer( $content ) {
+		$buffer_status = ob_get_status();
+		if ( ! empty( $buffer_status ) &&
+			1 === $buffer_status['type'] &&
+			get_class( $this ) . '::handle_buffer_content' == $buffer_status['name'] ) {
```

Comment:
👍🏻 I think this will be safer than a generic flush, though it could lead to edge cases where this gets skipped and the output buffer is not flushed at all. We should do some testing with different scenarios where a second buffer is started before and/or after ours, just in case.

Nit: use strict comparison.

```suggestion
			get_class( $this ) . '::handle_buffer_content' === $buffer_status['name'] ) {
```